teps/0135-coscheduling-pipelinerun-pods.md

```diff
@@ -137,12 +138,79 @@ PipelineRun pods can be scheduled to the same node using a mechanism similar to
+If the existing affinity assistant is enabled, the "coscheduling" flag must be set to "coschedule-workspaces". We will deprecate the "disable-affinity-assistant" flag in favor of the "coscheduling" flag, as discussed in ["Upgrade and Migration Strategy"](#upgrade-and-migration-strategy).
+
+The following chart summarizes the affinity assistant behaviors with different combinations of the "disable-affinity-assistant" and "coschedule" feature flags before and after the migration:
+
+<table>
+    <thead>
+        <tr>
+            <th>disabled-affinity-assistant</th>
+            <th>coscheduling</th>
+            <th>behavior during migration</th>
+            <th>behavior after migration</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr>
+            <td>false (default)</td>
+            <td>disabled</td>
+            <td>N/A: invalid</td>
+            <td>disabled</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>coschedule-workspaces (default)</td>
+            <td>coschedule workspaces</td>
+            <td>coschedule workspaces</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>coschedule-pipelineruns</td>
+            <td>N/A: invalid</td>
+            <td>coschedule pipelineruns</td>
+        </tr>
+        <tr>
+            <td>false (default)</td>
+            <td>isolate-pipelineruns</td>
+            <td>N/A: invalid</td>
+            <td>isolate pipelineruns</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>disabled</td>
+            <td>disabled</td>
+            <td>disabled</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>coschedule-workspaces (default)</td>
+            <td>disabled</td>
+            <td>coschedule workspaces</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>coschedule-pipelineruns</td>
+            <td>coschedule pipelineruns</td>
+            <td>coschedule pipelineruns</td>
+        </tr>
+        <tr>
+            <td>true</td>
+            <td>isolate-pipelineruns</td>
+            <td>isolate pipelineruns</td>
+            <td>isolate pipelineruns</td>
+        </tr>
+    </tbody>
+</table>
+
+This configuration preserves the default affinity assistant behavior ("coschedule workspaces") before and after the migration. During the migration, if users want to turn off affinity assistant, they can optionally set “coscheduling” to “disabled” so that removal of affinity assistant flag has no impact on them.
+
+One alternative is to use "disabled" as the default value for the "coscheduling" feature flag. However, it changes the default affinity assistant behavior to "disabled" after the migration. This change could introduce further complexity and confusion to users if we want to modify the default behavior to "coschedule-pipelineruns" based on users' feedback when promoting this feature to beta.
```

Comment:
Another con of this alternative is that it's not clear what the behavior should be if "disable-affinity-assistant" is "true" and "coscheduling" is set to "coschedule-workspaces".

Reply:
Updated!